WooCommerce/WooCommerceTests/ViewRelated/Dashboard/CardReaderSettings/CardReaderSettingsConnectedViewModelTests.swift

```diff
@@ -232,4 +233,66 @@ final class CardReaderSettingsConnectedViewModelTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
         XCTAssertFalse(analytics.receivedEvents.contains(WooAnalyticsStat.cardReaderSoftwareUpdateFailed.rawValue))
     }
+
+    func test_when_a_mandatory_update_succeeds_optional_updates_are_not_available() {
+        // Given
+        // .available is not sent
+        mockStoresManager.simulateUpdateStarted()
+        let expectation = self.expectation(description: #function)
```

Comment:
nit: we also have [a helper](https://github.com/woocommerce/woocommerce-ios/blob/dbea5342d2b1604904d109d6feb860b799b685d1/TestKit/Sources/TestKit/XCTestCase+Wait.swift#L72-L75) to wait for an expectation with a promise, you can find some examples in the codebase with `waitFor {` in the search. The helper makes it look a bit cleaner IMO 😄

Reply:
Thanks for the heads up on this!